Collections/Tree.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Tree.swift
+//  Collections
+//
+//  Created by James Bean on 12/9/16.
+//
+//
+
+/// Node in tree structure
+public enum Tree <T> {
+    
+    // MARK: - Cases
+    
+    /// Empty tree.
+    case empty
+    
+    /// Node with 0 or more subtrees.
+    indirect case node (T, [Tree])
```

Comment:
What is interesting though, is:

```Swift
enum Tree {
    case node (T,  [Tree])
}

throws up on itself until the stack blows up, but 

```Swift
enum Tree {
    case node ([Tree])
}
```
is perfectly fine.

Hmmm.